internal/ovirt-mini-api.go

```diff
@@ -75,9 +77,41 @@ func NewOvirt(configReader io.Reader) (OvirtApi, error) {
 	o.Connection.Password = viper.GetString("password")
 	o.Connection.Insecure = viper.GetBool("insecure")
 	o.Connection.CAFile = viper.GetString("cafile")
+
+	username, password := loadCredentials()
+	if username != "" {
+		o.Connection.Username = username
+	}
+	if password != "" {
+		o.Connection.Password = password
+	}
+
 	return &o, nil
 }
 
+func loadCredentials() (string, string) {
+	cred, exist := os.LookupEnv("OVIRT_CONNECTION_CREDENTIAL_FILE")
```

Comment:
Either we ensure we have a single source of configuration, and then read it all - like the conf file, or we decide we 
handle various conf options. You added both the bash code to write inject the credentials for flex, and now this here. 
I think its simpler do everthing in this method, and work without trying to inject stuff into the config. 

The first one wins:
- read env var for credentials
- read ovirt credentials file 
- read the regular conf file

And by that we would remain also backward compativle without any change the entrypoint and with 
knowing we never try to change the file.

Reply:
@rgolangh However, flexvolume driver container will have to copy the credentials over to the hostPath so that the kube-controller-manager can execute the driver with the correct configuration and credentials.

I would suggest the following:
* Streamline configuration and credentials into `ovirt-mini-api.go`:
    * from ovirt-provisioner.go
        * **deprecate**: Read from `/etc/ovirt/ovirt-api.conf` 
        * **deprecate**: Read from file specified with env var `OVIRT_API_CONF`
    * from ovirt-flexvolumedriver.go
       * **deprecate**: Read from file specified with env var `OVIRT_FLEXDRIVER_CONF`
       * **deprecate**: Read from current executable location + `/ovirt-flexvolume-driver.conf`
    * configuration - first one wins:
        * Read configuration from file specified with env var `OVIRT_CONF`
        * Read configuration from current executable location + `ovirt.conf`
        * Read configuration from `/etc/ovirt/ovirt.conf`
    * credentials - first one wins (overwrites `username` and `password` if previously specified in configuration file):
       * read username/password from env var `OVIRT_CONNECTION_USERNAME/PASSWORD`
        * read username/password from credentials file specified with `OVIRT_CONNECTION_CREDENTIAL_FILE`
        * read credentials from current executable location + `credentials.conf`


> And by that we would remain also backward compativle without any change the entrypoint and with knowing we never try to change the file.

It would still be necessary to adjust `entrypoint.sh` for:
* copying credentials file (either `OVIRT_CONNECTION_CREDENTIAL_FILE` or `credentials.conf`)
* inject username/password from `OVIRT_CONNECTION_USERNAME/PASSWORD`

To make it easier (only copying a credentials file) we could simply drop the env var credentials option.


Looking at the `entrypoint.sh`, would using the `OVIRT_FLEXDRIVER_CONF` env var to specify the config file currently even work? Since that one is not considered in the `entrypoint.sh` and thus would not be copied into the hostPath.


A more radical approach would be to:
* Only allow mounting of configuration/credentials file (no env variables involved): Then `entrypoint.sh` would only need to copy files from which it knows exactly where to locate them
* Otherwise we'd have to consider the other options and rewrite `entrypoint.sh`.


What's your view on this?

Reply:
@rgolangh To me there appears no other option than to edit `entrypoint.sh` since the kube-controller must have access to the credentials. Since we do not want to have all configuration set in a ConfigMap (Passwords/Usernames should be set in Secret) we somehow have to merge these locations together so that the ovirt-flexvolume-driver being executed under the kube-controller can read the configuration.

Another option would be to edit the `kube-controller` YAML file to set the environment variables. However, that adds additional complexity for the user to set up the plugin.

Since the ovirt-flexvolume-driver plugin code is executed after `entrypoint.sh` is executed and expects the configuration to be present, the `entrypoint.sh` must prepare the configuration file to be easily accessible by the driver (running from kube-controller which has no environment variables set with the information where the username/password is located).


Coming back to the original point made in this conversation (
https://github.com/oVirt/ovirt-openshift-extensions/pull/145#discussion-diff-323118467R93):
To reduce the complexity we could offer only the option to use only ENV variables
* OVIRT_CONNECTION_USERNAME
* OVIRT_CONNECTION_PASSWORD

and get rid of the file env variable option (https://github.com/oVirt/ovirt-openshift-extensions/pull/145/commits/cd9af906b375fa2c1f9ef002b0bfae703130a77a#diff-009c8c746dc10d6aa85738b5e4ae16b9R9).

Still `entrypoint.sh` must prepare the configuration file (adding username/password) so that ovirt-flexvolume-driver can access it when being run from `kube-controller-manager`.

Reply:
I'm fine with letting the entrypoint.sh prepare the the needed files or env vars.

Reply:
Thanks. I don't really see a better way with the FlexVolume approach.